runtime/lib/ttnn/program.cpp

```diff
@@ -868,11 +868,32 @@ static void run(::tt::target::ttnn::Conv2dOp const *op,
 static void run(::tt::target::ttnn::MaxPool2dOp const *op,
                 std::unordered_map<uint32_t, ::ttnn::Device *> &devicePool,
                 ProgramTensorPool &tensorPool) {
-  const ::ttnn::Tensor &input = tensorPool.at(op->in()->global_id());
+  ::ttnn::Tensor &input = tensorPool.at(op->in()->global_id());
   const ::ttnn::operations::pool::MaxPool2DOp operation =
       ::ttnn::operations::pool::MaxPool2DOp();
 
   ::ttnn::Device &device = getDevice(op->device(), devicePool);
+
+  // --------------- PRE-SHARDING ---------------
```

Comment:
Instead of making a comment block, perhaps we could have a local static function that does this?  @jnie-TT, has a PR that refactors max pool into its own file #713, so having a local func will even be more organized.